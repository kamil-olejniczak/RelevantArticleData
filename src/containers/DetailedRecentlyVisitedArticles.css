.DetailedRecentlyVisitedArticles {
    background-color: white;
    border-radius: 5px;
    padding: 10px;
    margin: 20px 0;
}

.DetailedRecentlyVisitedArticles a:visited {
    color: #00529B;
}