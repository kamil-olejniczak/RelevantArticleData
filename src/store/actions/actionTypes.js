export const INIT_GET_ARTICLE = 'INIT_GET_ARTICLE';
export const RESOLVING_ARTICLE = 'RESOLVING_ARTICLE';
export const GET_ARTICLE = 'GET_ARTICLE';
export const CLEAR_ARTICLE = 'CLEAR_ARTICLE';
export const INIT_RELOAD_ARTICLE = 'INIT_RELOAD_ARTICLE';
export const RELOAD_ARTICLE = 'RELOAD_ARTICLE';
export const INIT_CLEAR_DATA = 'INIT_CLEAR_DATA';
export const CLEAR_DATA = 'CLEAR_DATA';
export const INIT_SAVE_DATA = 'INIT_SAVE_DATA';
export const SAVE_DATA = 'SAVE_DATA';

export const INVALID_URL = 'INVALID_URL';
export const CLOSE_MODAL = 'CLOSE_MODAL';

export const ERROR_OCCURRED = 'ERROR_OCCURRED';
