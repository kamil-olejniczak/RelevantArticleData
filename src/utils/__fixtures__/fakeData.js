export const firstRecentlyVisitedSite = {
    id: 1,
    title: 'SomeTitle',
    url: 'SomeUrl',
    date: new Date(1 << 30),
};
export const secondRecentlyVisitedSite = {
    id: 2,
    title: 'SomeTitle2',
    url: 'SomeUrl2',
    date: new Date(1 << 30),
};
export const fakeRecentlyVisited = {
    recentlyVisited: [firstRecentlyVisitedSite, secondRecentlyVisitedSite]
};
export const fakeArticleData = {
    data: {
        title: 'Some Title',
        date_published: '1970-01-01',
        domain: 'www.dabliu.com',
        dek: 'Short dek text',
        excerpt: 'Short excerpt text',
        word_count: 111,
        lead_image_url: 'http://www.dabliu.pl/image.html',
    }
};
export const fakeTagStatistics = {
    div: 8,
    p: 2,
    a: 3,
    span: 10,
    i: 9,
    html: 1,
    br: 2,
    table: 7,
    hr: 6,
    input: 4,
    textarea: 5
};
export const fakeSortedTagStatistics = [
    'span', '10',
    'i', '9',
    'div', '8',
    'table', '7',
    'hr', '6',
    'textarea', '5',
    'input', '4',
    'a', '3',
    'br', '2',
    'p', '2',
];