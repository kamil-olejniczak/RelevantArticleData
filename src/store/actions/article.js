import * as actionTypes from './actionTypes';

export const articleResponse = ({data, content, recentlyVisited, statistics}, type) => ({
    type,
    data,
    content,
    recentlyVisited: recentlyVisited,
    statistics
});

export const getSpecifiedArticle = (url) => ({
    type: actionTypes.INIT_GET_ARTICLE,
    url
});

export const resolvingArticle = () => ({
    type: actionTypes.RESOLVING_ARTICLE
});

export const clearArticle = () => ({
    type: actionTypes.CLEAR_ARTICLE,
});

export const reloadPreviouslyVisitedArticle = (url) => ({
    type: actionTypes.INIT_RELOAD_ARTICLE,
    url
});

export const invalidUrl = () => ({
    type: actionTypes.INVALID_URL,
});

export const closeModalWithData = () => ({
    type: actionTypes.CLOSE_MODAL,
});

export const errorOccurred = (error) => ({
    type: actionTypes.ERROR_OCCURRED,
    error
});
