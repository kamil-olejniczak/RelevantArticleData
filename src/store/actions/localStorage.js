import * as actionTypes from './actionTypes';

export const clearData = () => ({
    type: actionTypes.INIT_CLEAR_DATA,
});

export const localStorageWasCleaned = () => ({
    type: actionTypes.CLEAR_DATA,
});

export const saveData = formattedData => ({
    type: actionTypes.INIT_SAVE_DATA,
    formattedData
});

export const localStorageWasUpdated = () => ({
    type: actionTypes.SAVE_DATA,
});