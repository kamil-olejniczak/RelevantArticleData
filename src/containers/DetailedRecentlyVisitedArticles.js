import React, {Component} from 'react';
import withNavbar from '../components/ui/navigation/Navbar';
import {connect} from 'react-redux';
import * as articleActions from '../store/actions/article';
import './DetailedRecentlyVisitedArticles.css';

export class DetailedRecentlyVisitedArticles extends Component {
    initReloadPreviouslyVisitedArticle = (url) => {
        this.props.reloadPreviouslyVisitedArticle(url);
        this.props.history.push('/');
    };

    render() {
        return (
            this.props.recentlyVisited.length ? (
                <div className="DetailedRecentlyVisitedArticles info">
                    {this.props.recentlyVisited.map((article, index) => (
                        <div key={article.id}>
                            <p>
                                <i className="fab fa-adn contentSeparator"></i>
                                Title: "{article.title}".
                            </p>
                            <p>
                                <i className="far fa-calendar-alt contentSeparator"></i>
                                Visited on: {article.date.toLocaleString()}.
                            </p>
                            <p>
                                <i className="fas fa-anchor contentSeparator"></i>
                                <span className="link"
                                      onClick={() => this.initReloadPreviouslyVisitedArticle(article.url)}>
                                    Reload article in our app.
                                </span>
                                <i className="fas fa-arrow-circle-up contentSeparator"></i><a href={article.url}>
                                Revisit this article on original page.</a>
                            </p>
                            {index !== this.props.recentlyVisited.length - 1 ? <hr/> : null}
                        </div>
                    ))}
                </div>
            ) : (
                <p className="warning">There is no data to be displayed, please search for an article.</p>
            )
        );
    }
}

const mapStateToProps = state => {
    return {
        recentlyVisited: state.article.recentlyVisited
    };
};

const mapDispatchToProps = dispatch => {
    return {
        reloadPreviouslyVisitedArticle: (url) => dispatch(articleActions.reloadPreviouslyVisitedArticle(url))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(withNavbar(DetailedRecentlyVisitedArticles));
