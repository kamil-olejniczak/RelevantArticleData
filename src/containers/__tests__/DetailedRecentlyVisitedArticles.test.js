import * as React from 'react';
import DetailedRecentlyVisitedArticles from '../DetailedRecentlyVisitedArticles';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {MemoryRouter} from 'react-router-dom';
import {mount, render} from 'enzyme';
import * as actionTypes from '../../store/actions/actionTypes';
import {fakeRecentlyVisited, firstRecentlyVisitedSite} from '../../utils/__fixtures__/fakeData';

describe('DetailedRecentlyVisitedArticles tests', () => {
    const mockStore = configureStore();

    let wrapper, store;

    it('renders correctly', () => {
        store = mockStore(fakeData);
        const component = renderer.create(componentWithRouter(store)).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should not render component when recentlyVisited array is empty', () => {
        store = mockStore({article: {recentlyVisited: []}});
        wrapper = render(componentWithRouter(store));

        expect(wrapper.find('.DetailedRecentlyVisitedArticles')).toHaveLength(0);
    });

    it('should render component when recentlyVisited array has proper data', () => {
        store = mockStore(fakeData);
        wrapper = render(componentWithRouter(store));

        expect(wrapper.find('.DetailedRecentlyVisitedArticles')).toHaveLength(1);
    });

    it('should not render horizontal line when there is only one element in recentlyVisited array', () => {
        store = mockStore({article: {recentlyVisited: [firstRecentlyVisitedSite]}});
        wrapper = render(componentWithRouter(store));

        expect(wrapper.find('hr')).toHaveLength(0);
    });

    it('should render horizontal line when there is more than one element in recentlyVisited array', () => {
        store = mockStore(fakeData);
        wrapper = render(componentWithRouter(store));

        expect(wrapper.find('hr')).toHaveLength(1);
    });

    it('should dispatch INIT_RELOAD_ARTICLE action from reloadPreviouslyVisitedArticle method', () => {
        store = mockStore(fakeData);
        wrapper = mount(componentWithRouter(store, {history: []}));

        wrapper.find('.link').first().simulate('click');

        expect(store.getActions()).toEqual([
            {type: actionTypes.INIT_RELOAD_ARTICLE, url: fakeData.article.recentlyVisited[0].url}
        ]);
    });
});

const fakeData = {
    article: {
        ...fakeRecentlyVisited
    }
};
const componentWithRouter = (store, props) => (
    <MemoryRouter>
        <DetailedRecentlyVisitedArticles store={store} {...props}/>
    </MemoryRouter>
);