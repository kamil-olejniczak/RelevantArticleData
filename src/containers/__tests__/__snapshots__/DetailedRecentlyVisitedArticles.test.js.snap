// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedRecentlyVisitedArticles tests renders correctly 1`] = `
<div>
  <ul
    className="Navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Search for an article
    </a>
    <a
      href="/recentlyVisited"
      onClick={[Function]}
    >
      Your articles
    </a>
    <li>
      <a
        href="http://github.com/kamil-olejniczak"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </li>
  </ul>
  <div
    className="DetailedRecentlyVisitedArticles info"
  >
    <div>
      <p>
        <i
          className="fab fa-adn contentSeparator"
        />
        Title: "
        SomeTitle
        ".
      </p>
      <p>
        <i
          className="far fa-calendar-alt contentSeparator"
        />
        Visited on: 
        1970-1-13 11:15:41
        .
      </p>
      <p>
        <i
          className="fas fa-anchor contentSeparator"
        />
        <span
          className="link"
          onClick={[Function]}
        >
          Reload article in our app.
        </span>
        <i
          className="fas fa-arrow-circle-up contentSeparator"
        />
        <a
          href="SomeUrl"
        >
          Revisit this article on original page.
        </a>
      </p>
      <hr />
    </div>
    <div>
      <p>
        <i
          className="fab fa-adn contentSeparator"
        />
        Title: "
        SomeTitle2
        ".
      </p>
      <p>
        <i
          className="far fa-calendar-alt contentSeparator"
        />
        Visited on: 
        1970-1-13 11:15:41
        .
      </p>
      <p>
        <i
          className="fas fa-anchor contentSeparator"
        />
        <span
          className="link"
          onClick={[Function]}
        >
          Reload article in our app.
        </span>
        <i
          className="fas fa-arrow-circle-up contentSeparator"
        />
        <a
          href="SomeUrl2"
        >
          Revisit this article on original page.
        </a>
      </p>
    </div>
  </div>
</div>
`;
